### ISSUE TITLE
test runner shows 'canceled' dialog

### ISSUE THREAD

**Issue by robaho** [oracle/javavscode#347]
Something has broken in a recent release.

When I run a Java test case, the following dialog appears and remains until a press the Cancel button. The test case runs fine, and the results appear correctly in the test tree.

<img width="874" alt="image" src="https://github.com/user-attachments/assets/34d8f79f-f60f-427f-8078-04213c347c71" />

I also get a notice in the lower right when this happens:

<img width="669" alt="image" src="https://github.com/user-attachments/assets/2091db07-1356-4c31-b914-8747450b528c" />

VSCode Version: 1.95.3 (Universal)
Java Extension version: v23.0.1
MacOS 13.7.1

**Comment by shivam71** [comment]
Thanks @robaho for pointing this out ! , you can resolve this issue for now by following the steps below :-

1. Navigate to the  [Run Debug View](https://code.visualstudio.com/docs/editor/debugging#_run-and-debug-view)  in the side panel.
2. Create a launch.json file for the project. [More on launch configurations](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations)
3. In case you don't see  any configuration object with attribute `"name": "Launch Java App"` then use  the   <br>  `Add Configuration...`  button appearing in the bottom right corner .  Among the list of options you get when you click the button select `Java+: Launch Java App`([Reference image](https://code.visualstudio.com/assets/docs/editor/debugging/launch-json-in-explorer.png)). Save the `launch.json` file.
4.  Now select the configuration named `Java+: Launch Java App` using the Configuration dropdown in the Run and Debug view . You should see `Java+: Launch Java App` in place of `Launch Program` with reference to this [image](https://code.visualstudio.com/assets/docs/editor/debugging/debugging_hero.png) .
5. Now proceed to run your test.

Currently this issue is arising because while running the test the configuration selected in the Run and Debug view is used  . (Perhaps in your case the `Java+: Attach to Process` was selected .) In future release we will fix this .

Let us know if this works for you .

**Comment by robaho** [comment]
I had the drop down set to 'Attach to Process'. Changed it to 'Launch Java App' and it worked. Thanks for the work-around.

**Comment by Achal1607** [comment]
Closing this issue due to inactivity. Please feel free to reopen this issue if you have any further queries.

**Comment by robaho** [comment]
I don’t think you should close the issue. I have a work around but this is less than ideal and should be fixed.
